internal/pkg/filtersBlockService/FilterMAIR.go

```diff
@@ -0,0 +1,166 @@
+// see: https://en.wikipedia.org/wiki/Interquartile_range
+package filters
+
+import (
+	"github.com/diadata-org/diadata/pkg/dia"
+	"github.com/diadata-org/diadata/pkg/model"
+	log "github.com/sirupsen/logrus"
+	"math"
+	"sort"
+	"strconv"
+	"time"
+)
+
+// FilterMAIR contains the configuration parameters of the filter
+type FilterMAIR struct {
+	symbol         string
+	exchange       string
+	currentTime    time.Time
+	previousPrices []float64
+	lastTrade      *dia.Trade
+	memory         int
+	value          float64
+}
+
+//NewFilterMAIR creates a FilterMAIR
+func NewFilterMAIR(symbol string, exchange string, currentTime time.Time, memory int) *FilterMAIR {
+	s := &FilterMAIR{
+		symbol:         symbol,
+		exchange:       exchange,
+		previousPrices: []float64{},
+		currentTime:    currentTime,
+		memory:         memory,
+	}
+	return s
+}
+func computeMean(samples []float64) (mean float64) {
+	var total float64
+	length := float64(len(samples))
+	if length == 0 {
+		return
+	}
+	for _, s := range samples {
+		total += s
+	}
+	mean = total / length
+	return
+}
+func computeMedian(samples []float64) (median float64) {
+	var length = len(samples)
+	if length > 0 {
+		sort.Float64s(samples)
+		if length%2 == 0 {
+			median = (samples[length/2-1] + samples[length/2]) / 2
+		} else {
+			median = samples[(length+1)/2-1]
+		}
+	}
+	return
+}
+func computeQuartiles(samples []float64) (Q1 float64, Q3 float64) {
+	sort.Float64s(samples)
+	var length = len(samples)
+	if length > 0 {
+		if length%2 == 0 {
+			Q1 = computeMedian(samples[0 : length/2])
+			Q3 = computeMedian(samples[length/2 : length])
+		} else {
+			Q1 = computeMedian(samples[0:int(math.Floor(float64(length/2)))])
+			Q3 = computeMedian(samples[int(math.Floor(float64(length/2)))+1 : length])
+		}
+	}
+	return
+}
+
+// RemoveOutliers Cleans a data set it accordance to the acceptable range within interquartile range.
+func removeOutliers(samples []float64) []float64 {
+	if len(samples) == 0 || len(samples) == 1 {
+		return samples
+	}
+	Q1, Q3 := computeQuartiles(samples)
+	IQR := Q3 - Q1
+	lowerBound := Q1 - 1.5*IQR
+	upperBound := Q3 + 1.5*IQR
+	lowerIndex := 0
+	upperIndex := len(samples)
+	for index, value := range samples {
+		if value < lowerBound {
+			lowerIndex = index + 1
+		} else if value > upperBound {
+			upperIndex = index
+			break
+		}
+	}
+	return samples[lowerIndex:upperIndex]
+}
+func (s *FilterMAIR) processDataPoint(price float64) {
+	/// first remove extra value from buffer if already full
+	if len(s.previousPrices) >= s.memory {
+		s.previousPrices = s.previousPrices[0 : s.memory-1]
+	}
+	s.previousPrices = append([]float64{price}, s.previousPrices...)
+}
+func (s *FilterMAIR) finalComputeEndOfBlock(t time.Time) {
+	if s.lastTrade == nil {
+		return
+	}
+	// Add the last trade again to compensate for the delay since measurement to EOB
+	// adopted behaviour from FilterMA
+	s.processDataPoint(s.lastTrade.EstimatedUSDPrice)
```

Comment:
I'm not sure we should do that, We do that with the MA because we use a "1 second" unit for the prices. if the price stays the same 5 seconds then it should weight stronger into the MA. What do you think @diamw @kaythxbye ?

Reply:
OK @mestorlx I had a chat with the elders of the internets, we will take your implementation.